### PR TITLE
Help: print real program name

### PR DESCRIPTION
**Pull request by g-maxime**
Signed-off-by: Maxime Gervais <gervais.maxime@gmail.com>

### CODE REVIEW

File: Source/CLI/CLI_Main.cpp [hunk 1]
```diff
@@ -117,6 +117,13 @@ int main(int argc, char* argv_ansi[])
     //Retrieve command line (mainly for Unicode)
     GETCOMMANDLINE();
 
+    //Get real program name, if aviable
+    if(argc && argv_ansi[0] != '\0')
+    {
+        String Program=argv[0];
+        Program_Name(Program);
```
Comment (JeromeMartinez) [inline Source/CLI/CLI_Main.cpp]: Why using intermediate Program variable?

File: Source/CLI/Help.cpp [hunk 2]
```diff
@@ -11,18 +11,53 @@
...
+            Program = Program.substr(Program.rfind('\\')+1);
+            Program = Program.substr(0, Program.find('.'));
+        #else
+            Program = Program.substr(Program.rfind('/')+1);
+        #endif
+    }
+
+    if(Program.empty())
+    {
+        #if defined(UNICODE)
```
Comment (JeromeMartinez) [inline Source/CLI/Help.cpp]: for all "UNICODE" stuff:
there is a "__T()" define, especially for not repeating code (is "L" with Unicode, "" with non Unicode)

File: Source/CLI/Help.cpp [hunk 3]
```diff
@@ -11,18 +11,53 @@
...
+    }
+
+    return Program;
+}
 
 //---------------------------------------------------------------------------
 int Help()
 {
-    TEXTOUT("Usage: \"MediaInfo [-Options...] FileName1 [Filename2...]\"");
+    String Program = Program_Name();
```
Comment (JeromeMartinez) [inline Source/CLI/Help.cpp]: No need of intermediate Program variable.

File: Source/CLI/Help.cpp [hunk 4]
```diff
@@ -11,18 +11,53 @@
...
+#include "Common/Core.h"
 #include "Help.h"
 #include "Config.h"
 //---------------------------------------------------------------------------
 
 //***************************************************************************
 //
 //***************************************************************************
+//---------------------------------------------------------------------------
+String Program_Name(const String &Set)
```
Comment (JeromeMartinez) [inline Source/CLI/Help.cpp]: Prefer to use a setter and a getter, not both in the same function (if Set is not empty, you don't need the returning String)

Comment (g-maxime) [inline Source/CLI/Help.cpp]: With two functions there will be a global variable

File: Source/CLI/Help.cpp (hunk 2 above)
Comment (g-maxime) [inline Source/CLI/Help.cpp]: tanks a lot!

File: Source/CLI/Help.cpp (hunk 4 above)
Comment (JeromeMartinez) [inline Source/CLI/Help.cpp]: Maybe, but I don't see how to do differently.
BTW, tested on Windows and it always shows the default value, looks normal as there is no glbal variable so argv[0] is not "saved". I don't see the idea behind the commit.

Comment (JeromeMartinez) [inline Source/CLI/Help.cpp]: lot of `String Program = Program_Name();`, which returns default value, but it should show argv[0].

**Comment by JeromeMartinez** [comment]
Closes https://github.com/MediaArea/MediaInfo/issues/163.
Testable with [snapshots](https://mediaarea.net/download/snapshots/binary/).

**Pull request merged**